freeipa/server/master.sls

```diff
@@ -12,7 +12,7 @@ freeipa_server_install:
         --domain {{ server.domain }}
         --hostname {% if server.hostname is defined %}{{ server.hostname }}{% else %}{{ grains['fqdn'] }}{% endif %}
         --ds-password {{ server.ldap.password }}
-        --admin-password {{ server.admin.password }}
+        --admin-password {{ salt['pillar.get']('server:admin:password', server.ldap.password) }}
```

Comment:
Same here, instead of salling pillar.get, use only server that you get from map.jinja